### PR TITLE
Remove powerfulMoveEffects array

### PR DESCRIPTION
**Pull request by AlexOn1ine**
The powerfulMoveEffects array is a Game Freak leftover that doesn't have much purpose at this point. The idea was that it should discourage certain moves but some of it doesn't make that much sense. E.g. Dream Eater should only be discouraged if the target isn't asleep or Solar Beam is just a normal move under Sun.

Already handled: 
``````
EFFECT_EXPLOSION
EFFECT_DREAM_EATER
``````
Created function `IsChargingMove` that checks if user holds a power herb or sun is active:
```
EFFECT_RECHARGE,
EFFECT_SKULL_BASH,
```

Moved to `AI_IsMoveEffectInMinus`:
```
EFFECT_SUPERPOWER
EFFECT_ERUPTION
EFFECT_OVERHEAT
EFFECT_MIND_BLOWN
EFFECT_MAKE_IT_RAIN
```

Also removed some more stuff that was handled by damage calcs.

### CODE REVIEW

File: src/battle_ai_main.c [hunk 1]
```diff
@@ -3163,13 +3143,15 @@ static s32 AI_CompareDamagingMoves(u32 battlerAtk, u32 battlerDef, u32 currId)
                 leastHits = noOfHits[i];
             }
             viableMoveScores[i] = AI_SCORE_DEFAULT;
-            isPowerfulIgnoredEffect[i] = IsInIgnoredPowerfulMoveEffects(gBattleMoves[moves[i]].effect);
+            isChargingMoveEffect[i] = IsChargingMove(battlerAtk, gBattleMoves[moves[i]].effect);
+            if (gBattleMoves[moves[i]].effect == EFFECT_METEOR_BEAM)
+                DebugPrintf("isChargingMoveEffect[i]: %d", isChargingMoveEffect[i]);
```
Comment (DizzyEggg) [inline src/battle_ai_main.c:3148]: Is this a testing leftover?

**Review by DizzyEggg — CHANGES_REQUESTED** [review]
Nice changes!

I would also add one new AI test which checks that Solarbeam is expected to use with Sun active, but otherwise it's discouraged.

**Comment by DizzyEggg** [comment]
Also.

This branch is out-of-date with the base branch
Merge the latest changes from upcoming into this branch.

File: src/battle_ai_main.c (hunk 1 above)
Comment (AlexOn1ine) [reply src/battle_ai_main.c:3148]: Yep

**Comment by AlexOn1ine** [comment]
> Nice changes!
> 
> I would also add one new AI test which checks that Solarbeam is expected to use with Sun active, but otherwise it's discouraged.

Would you say that recharge moves should always be discouraged if there is no Power Herb and in case of Solar Beam sun, or only if the number of hits is the same as other moves?

**Comment by DizzyEggg** [comment]
> > Nice changes!
> > I would also add one new AI test which checks that Solarbeam is expected to use with Sun active, but otherwise it's discouraged.
> 
> Would you say that recharge moves should always be discouraged if there is no Power Herb and in case of Solar Beam sun, or only if the number of hits is the same as other moves?

For now let's just consider the scenario where the number of hits is the same. In the future, we may think of some algorithms whether it's beneficial to use said moves or not.

**Comment by AlexOn1ine** [comment]
> Nice changes!
> 
> I would also add one new AI test which checks that Solarbeam is expected to use with Sun active, but otherwise it's discouraged.

Done. 

I used Knock Off to circumvent a problem where Solar Beam was expected turn 2 whenever it wasn't used turn 1. I guess it's not 100% accurate but I wanted to test that Solar Beam won't be chosen once Power Herb is gone.

**Review by DizzyEggg — APPROVED** [review]

**Pull request merged**